example/index/index.tf

```diff
@@ -0,0 +1,34 @@
+# Variables definition
+variable "url" {
+  type        = string
+  description = "The URL (connection string) of Couchbase server"
+}
+
+variable "username" {
+  type        = string
+  description = "A Couchbase user's name"
+}
+
+variable "password" {
+  type        = string
+  description = "A Couchbase user's password"
+}
+
+variable "bucket_creation_delay" {
+  type        = number
+  description = "A delay (in seconds) until the bucket is created on a cluster"
+}
+
+# A Couchbase provider
+provider "couchbase" {
+  url = var.url
+  username = var.username
+  password = var.password
+  bucket_creation_delay = var.bucket_creation_delay
+}
+
+# Create a primary index
+resource "couchbase_index" "index" {
+  bucket_name = "test"
+  index_name = "def_test_primary"
```

Comment:
`depends_on = [couchbase_bucket.bucket]` - it helped me to make it work together with bucket resource in one config

Reply:
better to use this `bucket_name = couchbase_bucket.bucket.name`, this allows making dependencies transparently

Reply:
yes, it's better